### PR TITLE
10046 staff payment

### PR DESCRIPTION
**Pull request by jordiwes**
*Issue #:* /bcgov/entity#10046

*Description of changes:*
- add staff payment dialog and button for staff
- another task is to "make it work"

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by doug-lovett — APPROVED** [review]
Looks good but some linter errors.

File: ppr-ui/src/components/dialogs/StaffPaymentDialog.vue [hunk 1]
```diff
@@ -0,0 +1,230 @@
+<template>
+  <v-dialog v-model="setDisplay" width="720px" persistent :attach="attach">
+    <v-card id="staff-payment-dialog" class="pl-5 pr-1 pt-5">
+      <v-row no-gutters>
+        <v-col cols="11">
+          <h3 class="dialog-title pb-6">Staff Payment</h3>
+          <div>
```
Comment (kialj876) [inline ppr-ui/src/components/dialogs/StaffPaymentDialog.vue]: Hey steph can you please update this to use the base dialog component? You can use slots to replace the body with the staff component / checkbox and props for the title / buttons. You can see I did something similar for the confirmation dialogs. This way all the formatting etc. will be the same / change with anything we alter in the future and we wont have to refactor this with the remaining dialogs

File: ppr-ui/tests/unit/StaffPaymentDialog.spec.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  it('renders with staff payment component', () => {
+    expect(wrapper.findComponent(StaffPaymentDialog).exists()).toBe(true)
+    expect(wrapper.findComponent(StaffPaymentComponent).exists()).toBe(true)
+    expect(wrapper.findAll(title).length).toBe(1)
+    expect(wrapper.find(title).text()).toBe('Staff Payment')
+    
+  })
+
+  it('Emits the cancel action', async () => {
```
Comment (kialj876) [inline ppr-ui/tests/unit/StaffPaymentDialog.spec.ts:66]: these will need to be tweaked to emit the action from the base dialog similar to the confirmation dialog tests

File: ppr-ui/tests/unit/StaffPaymentDialog.spec.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  afterEach(() => {
+    wrapper.destroy()
+  })
+
+  it('renders with staff payment component', () => {
+    expect(wrapper.findComponent(StaffPaymentDialog).exists()).toBe(true)
+    expect(wrapper.findComponent(StaffPaymentComponent).exists()).toBe(true)
+    expect(wrapper.findAll(title).length).toBe(1)
+    expect(wrapper.find(title).text()).toBe('Staff Payment')
+
```
Comment (kialj876) [inline ppr-ui/tests/unit/StaffPaymentDialog.spec.ts:63]: can you add another test in here to test that the staff payment comp is passed the correct data (i.e. check the props) and that the payment data gets successfully updated to the store and one for validity? I.e. emit the data from the staff component and check the values. There should also be a test in here that ensures the entered data gets cleared for when they do a second one :)

**Review by kialj876 — CHANGES_REQUESTED** [review]
Hey Steph, looks good I just have some comments I think are important. Let me know if you want to chat / have questions about anything :)

File: ppr-ui/tests/unit/StaffPaymentDialog.spec.ts (hunk 3 above)
Comment (jordiwes) [reply ppr-ui/tests/unit/StaffPaymentDialog.spec.ts:63]: My next task will be to make this dialog "work" :) ie save to store etc

File: ppr-ui/tests/unit/StaffPaymentDialog.spec.ts (hunk 2 above)
Comment (jordiwes) [reply ppr-ui/tests/unit/StaffPaymentDialog.spec.ts:66]: done

File: ppr-ui/src/components/dialogs/StaffPaymentDialog.vue (hunk 1 above)
Comment (jordiwes) [reply ppr-ui/src/components/dialogs/StaffPaymentDialog.vue]: done

**Review by kialj876 — APPROVED** [review]
Awesome, looks great 👍

**Pull request merged**